src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -440,12 +446,16 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
    * Handles deleting current feedback session.
    */
   deleteExistingSessionHandler(): void {
-    this.feedbackSessionsService.moveSessionToRecycleBin(this.courseId, this.feedbackSessionName).subscribe(() => {
-      this.navigationService.navigateWithSuccessMessage(this.router, '/web/instructor/sessions',
-          'The feedback session has been deleted. You can restore it from the deleted sessions table below.');
-    }, (resp: ErrorMessageOutput) => {
-      this.statusMessageService.showErrorToast(resp.error.message);
-    });
+    this.sessionEditFormModel.isDeleting = true;
+    this.feedbackSessionsService.moveSessionToRecycleBin(this.courseId, this.feedbackSessionName)
+     .pipe(finalize(() => this.sessionEditFormModel.isDeleting = false))
+     .subscribe(() => {
+       this.navigationService.navigateWithSuccessMessage(this.router, '/web/instructor/sessions',
+      'The feedback session has been deleted. You can restore it from the deleted sessions table below.');
+     },
+     (resp: ErrorMessageOutput) => {
+       this.statusMessageService.showErrorToast(resp.error.message);
+     });
```

Comment:
I have a minuscule doubt about the indentation here. The semantics doesn't seem to match up with the indentation that is :p

Reply:
I have fixed the link it now points to the correct issue.
I will throttle the page more and take another video of me pressing the save button so you can see what it did before my code was introduced.
Do you want me to change the indention? I fine with whatever you want. I tried to follow what I've seen in other parts of the code, but there is some inconsistency. I took my best guess and then made adjustments to get the code to pass the lint tests. Just let me know how it should look, and I'll make it happen.

Reply:
Here is the link to the requested video:
https://drive.google.com/file/d/1P6JeG2-OiGmEqb6_4hxVWj0tAVccttlW/view?usp=sharing

As soon as I press the save button it disappears and the edit button takes its place. The spinner doesn't show because the save button is removed before it appears. To fix it I changed the logic in the HTML file to allow the save button to stay until the save process finishes. (See after snapshots for proof: https://drive.google.com/file/d/1K_8a5zzV4VkIbe5xbap_vCa7EmGOwwu1/view?usp=sharing)

Reply:
I see. Indeed, this might be better in the UX.